### ISSUE TITLE
Standardizing HexModelGrid and RasterModelGrid init

### ISSUE THREAD

**Issue by kbarnhart** [landlab/landlab#864]
Right now the way that the two types of grids that use a "shape" are init is different.
This should be standardized in Landlab 2.0.

RasterModelGrid can take either rows and columns as individual arguments, or a shape tuple (rows, cols) for the first argument. For example:

```python
mg = RasterModelGrid(3,4)
```
vs 
```python
mg = RasterModelGrid((3,4))
```
HexModel grid, doesn't permit a shape tuple. 
```python
mg = HexModelGrid(3,4)
```
I don't have a strong feeling regarding which of these is better than the other. However, I do think that in Landlab 2.0 only one option should be supported and it should be consistent. If I had to choose, I'd lean towards providing a shape tuple. 

Tagging all, so that it can be discussed (and eventually I will make a PR deprecating the correct things).

@gregtucker @mcflugen @SiccarPoint @nicgaspar @nathanlyons @margauxmouchene @erkanistan @saisiddu

**Comment by gregtucker** [comment]
We definitely should be consistent among grid types (at least where it makes sense). I too lean toward having a shape tuple, simply because it is more consistent with Numpy arrays.

**Comment by mcflugen** [comment]
I also agree. The length of the tuple should be that of the number of grid dimensions. So, two for now but, when we introduce 1D (which we kind of already have) and 3D grids, the size of the tuple will range from one to three. This will make things easier as we won't have to add a new argument to the grid constructors.

I think the same should be done for radial grids. The *shape* tuple should be something that defines the grid in the *r* and *theta* dimensions.

**Comment by gregtucker** [comment]
Steps toward this:
- accept a tuple as 1st positional argument (and in that case handle second and third appropriately)
- accept node_layout as alternative keyword arg for ('hex' or 'rect')
- deprecate use of "shape" when it refers to a string instead of tuple

**Comment by kbarnhart** [comment]
Agree with @gregtucker 's steps. I'm turning the bullets into checks so we can keep track of what needs to be done.
